### DISCUSSION TITLE
My DSMR crashed and I need advise on how to restore🙋

### DISCUSSION THREAD

**Discussion by marco2152** [dsmrreader/dsmr-reader#1638] [Questions & answers / How-to's] [answered]
### Description

Mijn DSMR is rond 10 maart gecrashed en ik heb eigenlijk geen tijd gehad om daar naar te kijken. Tot nu toe. Ik wil DSMR weer aan de praat krijgen maar zie dat mijn sd-kaartje kapot is. Ik heb een nieuw kaartje met besturingssyteem gemaakt, mijn raspberry pi start op en ik heb beeld maar verder helemaal niks.
Ik neem aan dat ik opnieuw DSMR moet installeren en kan ik dan ook mijn laatste back-up terugzetten. Deze backup is van 10-3-2022. Graag advies hoe ik DSMR met backup weer aan de praat krijg.

Bedankt en groeten,
Marco

ps Is er al een manier om sd-kaartje te beschermen door bijv de data op een externe usb weg te schrijven of naar een database in de cloud oid? Ik denk dat dit nu de derde keer wordt om een sd-kaartje te vervangen en DSMR opnieuw te installeren. 

### DSMR-reader version

last version (at 10-3-2022)

**Comment by dennissiemensma** [comment, marked as answer]
Bedankt voor je vraag. Als je DSMR-reader en vooral de data echt wilt veiligstellen, dan zul je het niet op een SD kaart moeten draaien, maar op een server of iets met betrouwbare opslag (NAS?). 
Het is echt bedoeld als hobby project, waarbij je zelf de keuze maakt hoeveel tijd/geld/moeite je steekt in de borging van de gegevens. Je kunt het bijvoorbeeld prima op een VPS draaien en dan via de Pi gewoon de metingen doorsturen. Je kunt het ook op een thuisserver draaien, mits je dat hebt.

Het is echter allemaal niet triviaal, want je zult wel eea aan technische kennis moeten hebben hiervoor. Je kunt aan de Pi/server/NAS wel een niet-flash storage device hangen en dan via mounts de data-partitie van de database toewijzen aan de juiste storage. De kern staat hier:
- https://dsmr-reader.readthedocs.io/en/v5/how-to/database/postgresql-change-storage-location.html

----

Voor het terugzetten van de backup, kun je gewoon weer deze stappen volgen en tussen hoofdstuk 1 & 2 kun je je backup terugzetten:
- https://dsmr-reader.readthedocs.io/en/v5/tutorial/installation/step-by-step.html

**Reply by marco2152** [reply]
Je antwoord heeft geholpen, weer een werkende DSMR thuis. Bedankt! Ik wacht nog even op de overstap naar Docker totdat er een net zo'n geweldig stappenplan voorhanden is als op https://dsmr-reader.readthedocs.io/en/v5/tutorial/installation/step-by-step.html.
